### PR TITLE
Add Jenkinsfile.nightly to enable e2e test nightly

### PR DESCRIPTION
**Pull request by yiqian1**
https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/126

### CODE REVIEW

**Comment by whchung** [comment]
@jerryyin let's go over this PR in the weekly meeting shortly after. I'm about to add quite some amount of tests into E2E suite along with fix to fp16 overflow issue.

**Comment by jerryyin** [comment]
@asleepzzz I'd prefer to add back the SQLite test coverage which were commented out in PR #90. Wants to understand why it caused failure in your case, and how to properly fix it.

**Comment by asleepzzz** [comment]
I comment [-DMLIR_MIOPEN_SQLITE_ENABLED=1] due to if I don't add -DMIOPEN_SYSTEM_DB_PATH=`readlink -f ../mlir/utils/jenkins/perfdb/miopen_1.0.0.udb.resnet50_fp32_1013`  in cmake ,there are some lowering test failure,

but after I add the db path , failed in Dialect/MIOpen/Tuning/db_load_success.mlir 
If you can make sure check-mlir is ok ,just use it

File: mlir/utils/jenkins/Jenkinsfile.nightly [hunk 1]
```diff
@@ -0,0 +1,52 @@
+pipeline {
+    agent {
+        docker {
+            image 'rocm/mlir:rocm4.0-latest'
+            args '--user "$(id -u):$(id -g)" --device=/dev/kfd --device=/dev/dri --group-add video -u 0'
+            label 'rocm'
```
Comment (jerryyin) [inline mlir/utils/jenkins/Jenkinsfile.nightly:6]: Do we have machines that support xdlops in our CI?

File: mlir/utils/jenkins/Jenkinsfile.nightly [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                        -DLLVM_TARGETS_TO_BUILD="X86;AMDGPU" \
+                        -DCMAKE_BUILD_TYPE=Release \
+                        -DLLVM_ENABLE_ASSERTIONS=ON \
+                        -DBUILD_SHARED_LIBS=ON \
+                        -DLLVM_BUILD_LLVM_DYLIB=ON \
+                        -DMLIR_ROCM_RUNNER_ENABLED=1 \
+                        -DMLIR_MIOPEN_DRIVER_ENABLED=1 \
+                        -DMLIR_MIOPEN_DRIVER_E2E_TEST_ENABLED=1
+                        # if we open sqlite, step4 lowering failed until adding the custom db
+                        # -DMLIR_MIOPEN_SQLITE_ENABLED=1
```
Comment (jerryyin) [inline mlir/utils/jenkins/Jenkinsfile.nightly:36]: Could you re-enable this? This make sure we have test coverage on sqlite db read. We can talk to @asleepzzz if any specific failure come up.

**Review by jerryyin — CHANGES_REQUESTED** [review]
Two minor nits

File: mlir/utils/jenkins/Jenkinsfile.nightly (hunk 1 above)
Comment (yiqian1) [reply mlir/utils/jenkins/Jenkinsfile.nightly:6]: I don't think so, but I will verify with Omkar.

Comment (whchung) [reply mlir/utils/jenkins/Jenkinsfile.nightly:6]: we don't have one in CI cluster. the one we have is behind the firewall.

File: mlir/utils/jenkins/Jenkinsfile.nightly (hunk 2 above)
Comment (whchung) [reply mlir/utils/jenkins/Jenkinsfile.nightly:36]: I don't feel like enabling this in this PR, as the feature is really not used picking up any configs used by resnet50 or resnext101 as of now.

I disabled it because the flag caused more trouble than merit running `check-mlir` in local dev machine. It's extremely easy to get `check-mlir` to fail with sqlite on.

@jerryyin it'd be great if you could file a separate PR and work with stakeholders to ensure sqlite works end-to-end.

File: mlir/utils/jenkins/Jenkinsfile.nightly (hunk 1 above)
Comment (jerryyin) [reply mlir/utils/jenkins/Jenkinsfile.nightly:6]: I think we should push to add them in the CI. MI100 has already gone public, I don't see it necessary to be hidden in firewall anymore.

File: mlir/utils/jenkins/Jenkinsfile.nightly (hunk 2 above)
Comment (jerryyin) [reply mlir/utils/jenkins/Jenkinsfile.nightly:36]: Will do and leave a comment here.

File: mlir/utils/jenkins/Jenkinsfile.nightly (hunk 1 above)
Comment (jerryyin) [reply mlir/utils/jenkins/Jenkinsfile.nightly:6]: @yiqian1 When you clarify with Omkar, could you also ask what it takes to add MI100 node in our CI? Actually could you document this in an independent ticket so we can track the status there.

**Review by jerryyin — APPROVED** [review]

File: mlir/utils/jenkins/Jenkinsfile.nightly (hunk 2 above)
Comment (jerryyin) [reply mlir/utils/jenkins/Jenkinsfile.nightly:36]: Documented in https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/131
